llvm/lib/Transforms/Scalar/LoopInterchange.cpp

```diff
@@ -475,9 +481,33 @@ class LoopInterchangeLegality {
     return HasNoWrapReductions;
   }
 
+  // Record the simple reduction in the inner loop.
+  struct SimpleReduction {
+    // The reduction itself;
+    PHINode *Re;
+    // So far only supports constant initial value.
+    Value *Init;
+    Value *Next;
+    // The Lcssa PHI
+    PHINode *LcssaPhi;
+    // Only supports one user for now
+    // Store reduction result into memory object
+    StoreInst *LcssaStorer;
+    // The memory Location
+    Value *MemRef;
+    Type *ElemTy;
+
+    /// IV used for the loop exit condition.
+    PHINode *CounterIV;
+  };
+
+  const ArrayRef<SimpleReduction> getInnerSimpleReductions() const {
```

Comment:
```suggestion
  ArrayRef<SimpleReduction> getInnerSimpleReductions() const {
```

If I understand correctly, `ArrayRef` is immutable.